### PR TITLE
Use gai_strerror() to report errors of getaddrinfo() and getnameinfo()

### PR DESCRIPTION
**Pull request by masi**
Use gai_strerror() or strerror() depending on the return value of getaddrinfo() and getnameinfo() for a proper failure report.

See https://man.openbsd.org/gai_strerror.3

### CODE REVIEW

File: src/sock.c [hunk 1]
```diff
@@ -95,8 +95,13 @@ int opensock (const char *host, int port, const char *bind_to)
 
         n = getaddrinfo (host, portstr, &hints, &res);
         if (n != 0) {
+                const char *error;
```
Comment (rofl0r) [inline src/sock.c]: hmm, i think we should pack this into a function rather than duplicating the block further down

Comment (masi) [reply src/sock.c]: With a function like this?

```
const char * get_gai_error(int n) {
                if (n == EAI_SYSTEM)
                        return strerror (errno);
                else
                        return gai_strerror (n);
}
```

Comment (rofl0r) [reply src/sock.c]: right! but it should be static

File: src/sock.c [hunk 2]
```diff
@@ -95,13 +98,8 @@ int opensock (const char *host, int port, const char *bind_to)
...
         n = getaddrinfo (host, portstr, &hints, &res);
         if (n != 0) {
-                const char *error;
-                if (n == EAI_SYSTEM)
-                        error = strerror (errno);
-                else
-                        error = gai_strerror (n);
                 log_message (LOG_ERR,
-                             "opensock: Could not retrieve address info for %s:%d:", host, port, error);
+                             "opensock: Could not retrieve address info for %s:%d: ", host, port, get_gai_error (n));
```
Comment (rofl0r) [inline src/sock.c]: wouldnt we need an additional %s here ?

File: src/sock.c [hunk 3]
```diff
@@ -363,7 +354,7 @@ int getpeer_information (int fd, char *ipaddr, char *string_addr)
         ipaddr[0] = '\0';
         strlcpy (string_addr, "[unknown]", HOSTNAME_LENGTH);
 
-        /* Look up the IP address */
+        /* Look up the IP address */get_gai_error
```
Comment (rofl0r) [inline src/sock.c]: this artefact here should cause a build error

File: src/sock.c [hunk 4]
```diff
@@ -374,3 +365,14 @@ int getpeer_information (int fd, char *ipaddr, char *string_addr)
         return getnameinfo ((struct sockaddr *) &sa, salen,
                             string_addr, HOSTNAME_LENGTH, NULL, 0, 0);
 }
+
+/*
+ * Return a human readable error for getaddrinfo() and getnameinfo().
+ */
+static const char * get_gai_error (int n)
```
Comment (rofl0r) [inline src/sock.c]: cool. i guess though you need this function to the top so the callers can see the declaration

File: src/sock.c (hunk 3 above)
Comment (masi) [reply src/sock.c]: Sorry, for that.

File: src/sock.c (hunk 2 above)
Comment (masi) [reply src/sock.c]: Of course. Was missing since my first commit :(

File: src/sock.c (hunk 4 above)
Comment (masi) [reply src/sock.c]: Oh man, my C has rusted.

Comment (masi) [reply src/sock.c]: Is it possible to squash all those commits directly on Github? If not I'd better create another PR.

Comment (rofl0r) [reply src/sock.c]: yeah, you can squash them locally, then check everything's right with e.g. gitk (and a testbuild), then force-push

File: src/sock.c [hunk 5]
```diff
@@ -358,3 +378,4 @@ int getpeer_information (int fd, char *ipaddr, char *string_addr)
         return getnameinfo ((struct sockaddr *) &sa, salen,
                             string_addr, HOSTNAME_LENGTH, NULL, 0, 0);
 }
+
```
Comment (rofl0r) [inline src/sock.c]: looks good! apart from this gratuitous whitespace change here. wanna try the squashing yourself? if not i can do it too.

**Pull request merged**